### PR TITLE
Feat/allocation storage per period

### LINKED ISSUE #1367: Allocation storage per period
We want a new endpoint that takes a continuous range of blocks and returns the amount of data added during those blocks.

```sql
SELECT sum(size) from  WriteMarker where block_numer > start_block AND block_number < end_block
```

https://github.com/0chain/web-apps/issues/478

### PR DESCRIPTION
**Pull request by lawRathod**
## Changes
Closes https://github.com/0chain/0chain/issues/1367

## Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/0chain/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

## Associated PRs (Link as appropriate):
- blobber:
- gosdk:
- system_test:
- zboxcli:
- zwalletcli:
- Other: ...

### CODE REVIEW

File: code/go/0chain.net/smartcontract/dbs/event/writemarker.go [hunk 1]
```diff
@@ -55,6 +55,14 @@ func (edb *EventDb) GetAllocationWrittenSizeInLastNBlocks(blockNumber int64, all
 		Find(&total).Error
 }
 
+func (edb *EventDb) GetAllocationWrittenSizeInBlocks(startBlockNum, endBlockNum int64) (int64, error) {
+	var total int64
+	return total, edb.Store.Get().Model(&WriteMarker{}).
+		Select("sum(size)").
+		Where("block_number > ? AND block_number < ?", startBlockNum, endBlockNum).
```
Comment (dabasov) [inline code/go/0chain.net/smartcontract/dbs/event/writemarker.go:61]: add index to table

**Review by dabasov — CHANGES_REQUESTED** [review]

File: code/go/0chain.net/smartcontract/storagesc/handler.go [hunk 2]
```diff
@@ -78,6 +78,7 @@ func GetEndpoints(rh rest.RestHandlerI) []rest.Endpoint {
 		rest.MakeEndpoint(storage+"/getWritePoolAllocBlobberStat", srh.getWritePoolAllocBlobberStat),
 		rest.MakeEndpoint(storage+"/getChallengePoolStat", srh.getChallengePoolStat),
 		rest.MakeEndpoint(storage+"/alloc_written_size", srh.getWrittenAmount),
+		rest.MakeEndpoint(storage+"/allocWrittenSizePerPeriod", srh.getWrittenAmountPerPeriod),
```
Comment (peterlimg) [inline code/go/0chain.net/smartcontract/storagesc/handler.go]: Change to `/alloc-written-size-per-period`

**Review by peterlimg — DISMISSED** [review]

File: code/go/0chain.net/smartcontract/dbs/event/writemarker.go (hunk 1 above)
Comment (lawRathod) [reply code/go/0chain.net/smartcontract/dbs/event/writemarker.go:61]: Done

**Review by dabasov — APPROVED** [review]
lgtm

**Pull request merged**